File: java/src/main/java/dev/sanmer/sac/io/SacFileType.java
package dev.sanmer.sac.io;

public enum SacFileType {
    Time(1),
    RealImag(2),
    AmpPhase (3),
    XY(4);

    private final int iftype;

    SacFileType(int iftype) {
        this.iftype = iftype;
    }

    public int getIftype() {
        return iftype;
    }

    public static SacFileType valueBy(int iftype) {
        for (SacFileType type : SacFileType.values()) {
            if (type.iftype == iftype) {
                return type;
            }
        }

        throw new IllegalArgumentException();
    }
}

File: src/commonMain/kotlin/dev/sanmer/sac/io/SacFileType.kt
package dev.sanmer.sac.io

enum class SacFileType(val iftype: Int) {
    Time(1),
    RealImag(2),
    AmpPhase (3),
    XY(4);

    companion object {
        fun valueBy(iftype: Int) = when (iftype) {
            Time.iftype -> Time
            RealImag.iftype -> RealImag
            AmpPhase.iftype -> AmpPhase
            XY.iftype -> XY
            else -> throw IllegalArgumentException()
        }
    }
}